jerry-core/ecma/base/ecma-helpers-external-pointers.c

```diff
@@ -25,27 +25,35 @@
  */
 
 /**
- * Create internal property with specified identifier and store external pointer in the property.
- *
- * Note:
- *      property identifier should be one of the following:
- *        - LIT_INTERNAL_MAGIC_STRING_NATIVE_HANDLE
- *        - LIT_INTERNAL_MAGIC_STRING_FREE_CALLBACK
+ * Create a native handle property to store the native handle and its type info or free callback.
  *
  * @return true - if property was just created with specified value,
  *         false - otherwise, if property existed before the call, it's value was updated
  */
 bool
-ecma_create_external_pointer_property (ecma_object_t *obj_p, /**< object to create property in */
-                                       lit_magic_string_id_t id, /**< identifier of internal
-                                                                  *   property to create */
-                                       ecma_external_pointer_t ptr_value) /**< value to store in the property */
+ecma_create_native_handle_property (ecma_object_t *obj_p, /**< object to create property in */
+                                    ecma_external_pointer_t handle_p, /**< native handle */
+                                    ecma_external_pointer_t info_p, /**< native handle's type info */
+                                    ecma_external_pointer_t free_cb) /**< native handle's free callback*/
 {
-  JERRY_ASSERT (id == LIT_INTERNAL_MAGIC_STRING_NATIVE_HANDLE
-                || id == LIT_INTERNAL_MAGIC_STRING_FREE_CALLBACK);
+  bool is_deprecate = (info_p == (ecma_external_pointer_t) NULL);
+  void *package_p;
```

Comment:
Can't this be `ecma_native_handle_package_t *package_p` and save the casting in lines 52-53?